internal/test/testdata/verify/cases/case_011.yaml.input

```diff
@@ -0,0 +1,176 @@
+-- testdata/principals.yaml --
+---
+principals:
+  john:
+    id: john
+    roles:
+      - employee
+    attr:
+      department: marketing
+      geography: GB
+      team: design
+  bev:
+    id: bev
+    roles:
+      - employee
+      - manager
+    attr:
+      department: marketing
+      geography: GB
+      managed_geographies: GB
+      team: design
+  terry_tibbs:
+    id: terry_tibbs
+    roles:
+      - employee
+    attr:
+      department: accounting
+      geography: GB
+      team: payroll
+
+-- testdata/resources.yaml --
+---
+resources:
+  john_equipment_request:
+    kind: equipment_request
+    id: XX125
+    scope: acme
+    attr: &attr
+      department: marketing
+      geography: GB
+      id: XX125
+      owner: john
+      team: design
+  john_pending_equipment_request:
+    kind: equipment_request
+    id: XX225
+    scope: acme
+    attr:
+      << : *attr
+      status: PENDING_APPROVAL
+
+-- suite_test.yaml --
+---
+name: TestSuite
+description: Tests for equipment requests
+tests:
+  - name: John's request
+    input:
+      principals:
+        - john
+        - bev
+        - terry_tibbs
+      resources:
+        - john_equipment_request
+        - john_pending_equipment_request
+      actions:
+        - view:public
+        - approve
+        - create
+    expected:
+      - principal: john
+        resource: john_equipment_request
+        actions:
+          view:public: EFFECT_ALLOW
+          approve: EFFECT_DENY
+          create: EFFECT_ALLOW
+        outputs:
+          - action: view:public
+            expected:
+              - src: "resource.equipment_request.vdefault#public-view"
+                val:
+                  id: john
+                  keys: XX125
+                  formatted_string: "id:john"
+                  some_bool: true
+                  some_list: ["foo", "bar"]
+                  something_nested:
+                    nested_str: foo
+                    nested_bool: false
+                    nested_list: ["nest_foo", 1.01]
+                    nested_formatted_string: "id:john"
```

Comment:
This is more food for thought:

In this instance (by design), we're rejecting `approve` because the request doesn't have the [appropriate status](https://github.com/Sambigeara/cerbos/blob/97dc68926c2d564d8d3c2354d7ff760656bb26a1/internal/test/testdata/store/resource_policies/policy_07.yaml#L32-L41), but not returning the output.

I wonder if the output would be valuable here, because you'd be able to tell at a glance the reason for DENY (rather than by inspecting the policies afterwards to match).

Reply:
I see your point but `output` is not a debugging aid. It's a signal to the API user that a particular rule was activated (all the conditions matched) during the evaluation so that they can take extra steps based on that information. If we produce outputs for all rules that were considered but not activated, then that signal is lost.

For debugging purposes we have the engine traces and the REPL.